packages/shared/src/components/header/FeedExploreHeader.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import { BreadCrumbs } from './BreadCrumbs';
+import { CalendarIcon, HotIcon } from '../icons';
+import { IconSize } from '../Icon';
+import TabList from '../tabs/TabList';
+import { Tab, TabContainer } from '../tabs/TabContainer';
+import { checkIsExtension } from '../../lib/func';
+import { SharedFeedPage } from '../utilities';
+import { getFeedName } from '../../lib/feed';
+import { Dropdown } from '../fields/Dropdown';
+import { QueryStateKeys, useQueryState } from '../../hooks/utils/useQueryState';
+import { periodTexts } from '../layout/common';
+
+export enum ExploreTabs {
+  Popular = 'Popular',
+  MostUpvoted = 'By upvotes',
+  BestDiscussions = 'By comments',
+  ByDate = 'By date',
+}
+
+const tabsToFeedMap: Partial<Record<SharedFeedPage, ExploreTabs>> = {
+  [SharedFeedPage.Explore]: ExploreTabs.Popular,
+  [SharedFeedPage.ExploreUpvoted]: ExploreTabs.MostUpvoted,
+  [SharedFeedPage.ExploreDiscussed]: ExploreTabs.BestDiscussions,
+  [SharedFeedPage.ExploreLatest]: ExploreTabs.ByDate,
+};
+
+const urlToTab: Record<string, ExploreTabs> = {
+  '/explore': ExploreTabs.Popular,
+  '/explore/upvoted': ExploreTabs.MostUpvoted,
+  '/explore/discussed': ExploreTabs.BestDiscussions,
+  '/explore/latest': ExploreTabs.ByDate,
+};
+
+interface FeedExploreHeaderProps {
+  tab: ExploreTabs;
+  setTab: (tab: ExploreTabs) => void;
+}
+
+const withDateRange = [
+  SharedFeedPage.ExploreUpvoted,
+  SharedFeedPage.ExploreDiscussed,
+];
+
+export function FeedExploreHeader({
+  tab,
+  setTab,
+}: FeedExploreHeaderProps): ReactElement {
+  const isExtension = checkIsExtension();
+  const router = useRouter();
+  const path = getFeedName(router.pathname);
+  const [period, setPeriod] = useQueryState({
+    key: [QueryStateKeys.FeedPeriod],
+    defaultValue: 0,
+  });
+
+  return (
+    <div className="flex w-full flex-col">
+      <BreadCrumbs className="px-2">
+        <HotIcon size={IconSize.XSmall} secondary /> Explore
+      </BreadCrumbs>
+      <div className="my-4 flex flex-row items-center">
+        {isExtension ? (
+          <TabList<ExploreTabs>
+            items={Object.values(ExploreTabs)}
+            active={tab}
+            onClick={setTab}
+          />
+        ) : (
+          <TabContainer
+            controlledActive={tabsToFeedMap[path]}
+            className={{ header: 'border-b-0' }}
+            shouldMountInactive
+          >
+            {Object.entries(urlToTab).map(([url, label]) => (
+              <Tab key={label} label={label} url={url} />
+            ))}
+          </TabContainer>
```

Comment:
This isn't something we do with all tabs we have extension/web app? would it be worth this was part of tab components to differentiate between both depending on where it's used?

Reply:
The extension and webapp would behave differently when the tabs are clicked. The `TabContainer` has a built-in function to do redirection. I would've used `TabList` on both of them if it had supported the redirection as mentioned.

Reply:
Well we're seeing them in one file or another 😅 Anyways I think it's ok, we can refactor later if another thing like this comes our way

Reply:
Yeah, I think we can clean this up once we decide to replace the Popular page with the new Explore page eventually. I mean as this is technically an experiment.